### PR TITLE
New Security Considerations

### LINKED ISSUE #9: Effect on stateless resets
The draft doesn't address the impact of each method of connection ID generation on how servers can use stateless resets.

Most of this is likely bound up in decisions stemming from #8.  If you can guess a valid but unused connection ID, then you might be able to induce a stateless reset that could be used to kill an open connection.

As the draft only includes methods that include an explicit server identifier, it is possible that as long as valid values cannot be guessed, the effect is minimal and each server instance can have its own configured stateless reset key (or a shared key from which a per-server key is derived using a KDF).

### PR DESCRIPTION
**Pull request by martinduke**
Added two sections to Security Considerations (Stateless Reset Oracle and Local Configurations Only). Fixes #9 and #27.

### CODE REVIEW

File: draft-ietf-quic-load-balancers.md [hunk 1]
```diff
@@ -926,6 +926,27 @@ connection IDs to the same server.  The QUIC-LB algorithms do prevent the
 linkage of two connection IDs to the same individual connection if servers make
 reasonable selections when generating new IDs for that connection.
 
+## Limited configuration deployment
+
+A naive deployment of QUIC-LB in a cloud provider might use the same global
+QUIC-LB configuration across all the load balancers in its enterprise. An
```
Comment (martinthomson) [inline draft-ietf-quic-load-balancers.md:932]: I don't think that the scope of this is very clear.  If the idea is that the cloud provider - the one who provides the load balancer - provides the same configuration to mutually distrustful customers, then there is a very important point to make.  But I'm not sure that this is clear from the text here.  "enterprise" isn't well connected to "cloud provider".

Comment (martinduke) [reply draft-ietf-quic-load-balancers.md:932]: OK, I got rid of the word 'enterprise' and I think it's clearer now.

File: draft-ietf-quic-load-balancers.md [hunk 2]
```diff
@@ -926,6 +926,28 @@ connection IDs to the same server.  The QUIC-LB algorithms do prevent the
...
+
+A naive deployment of QUIC-LB in a cloud provider might use the same global
+QUIC-LB configuration across all its load balancers that route to customer
+servers. An attacker could then simply become a customer, obtain the
+configuration, and then extract server IDs of other customers' connections at
+will.
+
+To avoid this, the entities that manage load balancers that route to mutually
+distrustful servers SHOULD use a single configuration instance in as narrow a
+deployment as possible. Ideally, a load balancer would not have any QUIC-LB
```
[the remark this answers is not in the document]
Comment (martinduke) [reply draft-ietf-quic-load-balancers.md]: OK, I've rephrased it to be a bit more restrictive. How do you like it?

**Comment by martinthomson** [comment]
Can you speak a little about your reasoning for the recent changes?  If this attack is limited to extracting the identity of server instances for other co-hosted entities, that's probably OK, but I'm having trouble connecting the SHOULD here with the preceding text.

**Comment by martinduke** [comment]
Sure. If an attacker has the same QUIC-LB config as the victim, then it can extract the server mapping, which defeats the whole point of the spec. Obviously, an LB must have the QUIC-LB config for servers it routes to. If I'm an attacker within that group of servers I may already see the packet headers that give away the server mapping, so there's little added linkability here (though there is a little).

The moment I share configuration among multiple server pools, I'm expanding the number of entities with access to the config, for no operational benefit except to make the administrator's life easier. In the absurd limit, all of AWS has the same QUIC-LB config, and essentially everyone can extract the server mapping from any CID that goes to AWS.

**Pull request merged**